tests/smoke/alp_cholesky.cpp

```diff
@@ -19,60 +19,121 @@
+	return static_cast< BaseScalarType >( rand() ) / RAND_MAX;
+}
+
+template<>
+std::complex< BaseScalarType > random_value< std::complex< BaseScalarType > >() {
+	const BaseScalarType re = random_value< BaseScalarType >();
+	const BaseScalarType im = random_value< BaseScalarType >();
+	return std::complex< BaseScalarType >( re, im );
+}
+
+
 struct inpdata {
 	std::string fname="";
 	size_t N=0;
 };
 
-//** gnerate upper/lower triangular part of a SPD matrix */
+
+/** Generate full storage Symmetric or Hermitian
+ *   positive definite matrix for in-place tests
+ */
 template< typename T >
-void generate_spd_matrix_full( size_t N, std::vector<T> &data ) {
+void  generate_symmherm_pos_def_mat_data_full(
+	size_t N,
+	std::vector< T > &mat_data
+) {
+	std::fill( mat_data.begin(), mat_data.end(), static_cast< T >( 0 ) );
 	for( size_t i = 0; i < N; ++i ) {
-		for( size_t j = 0; j < N; ++j ) {
-			size_t k = i * N + j;
-			if( i <= j ) {
-				data[ k ] = static_cast< T >( std::rand() ) / static_cast< T >( RAND_MAX );
-			}
+		for( size_t j = i; j < N; ++j ) {
+			mat_data[ i * N + j ] = random_value< T >();
+			mat_data[ j * N + i ] += grb::utils::is_complex< T >::conjugate( mat_data[ i * N + j ] );
 			if( i == j ) {
-				data[ k ] = data[ k ] + static_cast< T >( N );
+				mat_data[ j * N + i ] += static_cast< T >( N );
 			}
-			if( i > j ) {
-				data[ i * N + j ] = data[ j * N + i ];
-			}
-
 		}
 	}
 }
 
+/** Generate symmetric-hermitian positive
+ *  definite matrix in a full-storage container
+ */
+template< typename T >
+void generate_symmherm_pos_def_mat_data(
+	size_t N,
+	std::vector< T > &mat_data,
+	const typename std::enable_if<
+		grb::utils::is_complex< T >::value,
+		void
+	>::type * const = nullptr
+) {
+	generate_symmherm_pos_def_mat_data_full< T >( N, mat_data );
+}
 
-//** gnerate upper/lower triangular part of a SPD matrix */
+/** Generate upper/lower triangular part of a
+ *  Symmetric positive definite matrix
+ */
 template< typename T >
-void generate_spd_matrix( size_t N, std::vector<T> &data ) {
+void  generate_symmherm_pos_def_mat_data(
+	size_t N,
+	std::vector< T > &mat_data,
+	const typename std::enable_if<
+		!grb::utils::is_complex< T >::value,
+		void
+	>::type * const = nullptr
+) {
+	std::fill( mat_data.begin(), mat_data.end(), static_cast< T >( 0 ) );
 	size_t k = 0;
 	for( size_t i = 0; i < N; ++i ) {
 		for( size_t j = i; j < N; ++j ) {
-			data[ k ] = static_cast< T >( std::rand() ) / static_cast< T >( RAND_MAX );
+			mat_data[ k ] = random_value< T >();
 			if( i == j ) {
-				data[ k ] = data[ k ] + static_cast< T >( N );
+				mat_data[ k ] += grb::utils::is_complex< T >::conjugate( mat_data[ k ] );
+				mat_data[ k ] += static_cast< T >( N );
 			}
 			++k;
 		}
 	}
 }
 
 
-//** check the solution by calculating the Frobenius norm of (H-LL^T) */
+/** Check the solution by calculating the
+ *  Frobenius norm of (H-LL^T)
+ */
 template<
 	typename MatSymmType,
 	typename MatUpTriangType,
```

Comment:
I would suggest to start fixing the (matrix name) notation L -> U if not everywhere at least at the top within this smoke test. Just to avoid throwing off someone interested in checking how this problem is specified at the test level.

Reply:
Done!